mobile/src/modules/auth/store/actions.ts

```diff
@@ -81,12 +82,26 @@ export const createOfflineAccount = createActionFn<string, Account>(
     }
 
     // TODO: make fields optional in @burst package
-    // @ts-ignore
-    return {
+    return new Account({
       type: 'offline', // TODO: make type enum in @burst package
       account,
       accountRS
-    };
+    });
+  }
+);
+
+export const hydrateAccount = createActionFn<Account, Promise<Account>>(
+  async (dispatch, getState, account) => {
+
+    const api = composeApi(new ApiSettings('https://wallet1.burst-team.us:2083', 'burst'));
+    try {
+      const accountDetails = await api.account.getAccount(account.account);
+      dispatch(actions.updateAccount(accountDetails));
+    // @ts-ignore
+    } catch (e) { }
+
+    await setAccounts(getState().auth.accounts);
```

Comment:
In redux world often so called selectors are used to access fragments of the state, or even return computed values. 
`export const selectAccounts = state => state.auth.accounts`

And you can use them also  in `mapStateToProps` ...